ballista/executor/src/executor_process.rs

```diff
@@ -181,11 +182,25 @@ pub async fn start_executor_process(opt: ExecutorProcessConfig) -> Result<()> {
         BallistaError::Internal("Failed to init Executor RuntimeEnv".to_owned())
     })?);
 
+    let mut service_handlers: FuturesUnordered<JoinHandle<Result<(), BallistaError>>> =
+        FuturesUnordered::new();
+
     let metrics_collector = Arc::new(LoggingMetricsCollector::default());
 
+    let (replicator_send, replicator_recv) = mpsc::channel::<replicator::Command>(10);
+    let replication_url =
+        ObjectStoreUrl::parse("authority").expect("Invalid default catalog url");
+
+    service_handlers.push(tokio::spawn(replicator::start_replication(
+        "",
+        runtime.object_store(replication_url)?,
+        replicator_recv,
+    )));
+
     let executor = Arc::new(Executor::new(
         executor_meta,
         &work_dir,
+        Some(replicator_send.clone()),
```

Comment:
cloning to leave one sender alive to keep the channel open for a while lifetime of exec until termination